packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -82,25 +98,100 @@ export class EditorSettingsSidebarComponent {
 	 * @returns {Promise<void>} No return value.
 	 */
 	async clickTab( tabName: EditorSidebarTab ): Promise< void > {
-		await this.frame.click( selectors.tabButton( tabName ) );
-		await this.frame.waitForSelector( selectors.activeTabButton( tabName ) );
+		const locator = this.frameLocator.locator( selectors.tabButton( tabName ) );
+		await locator.click();
+
+		const classAttributes = await locator.getAttribute( 'class' );
+		if ( ! classAttributes?.includes( 'is-active' ) ) {
+			throw new Error( `Failed to verify ${ tabName } is active.` );
+		}
```

Comment:
While I really like the simplicity of this, I'm a little bit nervous about race conditions herer, which was the original reason for using the `waitFor` combined with a selector that identifies the active state.

When you click on a tab, it's async updating the React state, which is then re-rendering that tab with the new class as calculated from that state. While that should be lightening fast, Playwright is also really fast, and `getAttribute` returns immediately. 

It's definitely possible that this race condition will never occur in reality! But I feel like we've seen similar things happen, and it makes me a little nervous.

Thoughts?

Reply:
Replaced this logic with the following:

```
		const activeTabLocator = this.frameLocator.locator(
			`${ selectors.tabButton( tabName ) }.is-active`
		);
		await activeTabLocator.waitFor();
```